gettsim/benefits/kiz.py

```diff
@@ -68,36 +67,37 @@ def kiz(household, params, arbeitsl_geld_2_params, kindergeld_params):
     #        plus Gesamtkinderzuschlag liegen.
     #     3. Dann wird geschaut, wie viel von dem Einkommen
     #        (Erwachsene UND Kinder !) noch auf KIZ angerechnet wird.
-    #        Wenn das zu berücksichtigende Einkommen UNTER der
+    #        Wenn das zu berücksichtigende Einkommen der Eltern UNTER der
     #        Höchsteinkommensgrenze und UNTER der Bemessungsgrundlage liegt, wird
     #        der volle KIZ gezahlt
     #        Wenn es ÜBER der Bemessungsgrundlage liegt,
-    #        wird die Differenz zur Hälfte abgezogen.
+    #        wird die Differenz zu einem gewissen Anteil abgezogen.
     household["kiz_ek_gross"] = household["alg2_grossek_hh"]
     household["kiz_ek_net"] = household["ar_alg2_ek_hh"]
 
-    # Deductable income. 50% withdrawal rate.
-    household["kiz_ek_anr"] = np.maximum(
-        0, 0.5 * (household["ar_alg2_ek_hh"] - household["kiz_ek_relev"])
+    # 1st step: deduct children income for each eligible child (§6a (3) S.3 BKGG)
+    household["kiz_childinc_deducted"] = household["child_kg"] * (
+        np.maximum(0, params["a2kiz"] - (household["m_wage"] + household["uhv"]))
     )
 
+    # 2nd step: Calculate the parents income that needs to be subtracted
+    # (§6a (6) S. 3 BKGG)
+    household["kiz_ek_anr"] = np.maximum(
+        0,
+        params["a2kiz_withdrawal_rate"]
+        * (household["ar_alg2_ek_hh"] - household["kiz_ek_relev"]),
+    )
     # Dummy variable whether household is in the relevant income range.
     household["kiz_incrange"] = (
         household["kiz_ek_gross"] >= household["kiz_ek_min"]
     ) & (household["kiz_ek_net"] <= household["kiz_ek_max"])
```

Comment:
Is there a reason not to use `alg2_grossek_hh` and `ar_alg2_ek_hh` directly here? I see no other occurrences of `kiz_ek_gross` and `kiz_ek_net` anywhere.

In that case you can drop:
```python
    household["kiz_ek_gross"] = household["alg2_grossek_hh"]
    household["kiz_ek_net"] = household["ar_alg2_ek_hh"]
```

Reply:
This is how it was done before. I find it way more intuitive this way.